src/components/WwwFrame/PromotionalBanner/Banners/AppealBanner/AppealBanner.vue

```diff
@@ -0,0 +1,344 @@
+								:open="open"
+							/>
+						</div>
+						<div class="small-12 medium-9 large-8 xlarge-7 columns">
+							<div>
+								<span
+									v-html="bannerBody"
+									class="message"
+								></span>
+								<!-- <span class="strong" v-html="bannerMatchingText"></span> -->
+							</div>
+							<div class="show-for-small hide-for-medium small-12 columns thermometer-holder">
+								<appeal-thermometer
+									:percent-toward-goal="percentTowardGoal"
+								/>
+							</div>
+							<div class="donation-buttons">
+								<ul>
+									<li v-for="(buttonAmount, index) in buttonAmounts"
+										:key="index"
+									>
+										<kv-button
+											class="mini custom-width"
+											@click.native.prevent.stop="updateDonationTo(buttonAmount)"
+										>
+											${{ buttonAmount }}
+										</kv-button>
+									</li>
+								</ul>
+								<a
+									class="other-amount"
+									href="/donate/supportus"
+									@blur="validateInput"
+									v-kv-track-event="['promo', 'click', 'EOYBanner', 'other-button']"
+								>Other amount
+								</a>
+							</div>
+						</div>
+					</div>
+				</div>
+			</kv-expandable>
+		</div>
+	</div>
+</template>
+
+<script>
+import numeral from 'numeral';
+import _forEach from 'lodash/forEach';
+import _get from 'lodash/get';
+import store2 from 'store2';
+import AppealThermometer from '@/components/WwwFrame/PromotionalBanner/Banners/AppealBanner/AppealThermometer';
+import KvButton from '@/components/Kv/KvButton';
+import KvIcon from '@/components/Kv/KvIcon';
+import appealBannerQuery from '@/graphql/query/appealBanner.graphql';
+import KvExpandable from '@/components/Kv/KvExpandable';
+import updateDonation from '@/graphql/mutation/updateDonation.graphql';
+import { documentToHtmlString } from '~/@contentful/rich-text-html-renderer';
+
+export default {
+	components: {
+		KvButton,
+		KvIcon,
+		AppealThermometer,
+		KvExpandable,
+	},
+	inject: ['apollo'],
+	props: {
+		appealMatchEnabled: {
+			type: Boolean,
+			default: false,
+		},
+		appealBannerContent: {
+			type: Object,
+			default: () => {},
+		}
+	},
+	data() {
+		return {
+			open: true,
+			amount: 0,
+			donationAmount: null,
+			percentTowardGoal: null,
+		};
+	},
+	apollo: {
+		query: appealBannerQuery,
+		preFetch: true,
+		result({ data }) {
+			this.amountRaised = _get(data, 'general.kivaStats.latestDonationCampaign.amount_raised');
+			this.targetAmount = _get(data, 'general.kivaStats.latestDonationCampaign.target_amount');
+		},
+	},
+	computed: {
+		bannerHeadline(props) {
+			const appealHeadline = props.appealBannerContent.richText;
+			return documentToHtmlString(appealHeadline);
+		},
+		bannerBody(props) {
+			const appealBody = props.appealBannerContent.additionalContent[0].fields.richText;
+			return documentToHtmlString(appealBody);
+		},
+		buttonAmounts(props) {
+			const { donationAmounts } = props.appealBannerContent.dataObject;
+			return donationAmounts;
```

Comment:
Please check for donation amounts and provide defaults if the donationAmounts array is empty. You can set the defaults to the current 3 options.

Reply:
Put in an attempt to fix this, though I'm having issues figuring out the right check to put in place. Going to eat some food and take a break before revisiting.